presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -113,18 +113,20 @@
     private final ThriftHiveMetastoreStats stats;
     private final HiveCluster clientProvider;
     private final Function<Exception, Exception> exceptionMapper;
+    private final ThriftHiveMetastoreConfig thriftConfig;
```

Comment:
Config classes are mutable, so we never store them as fields (in theory, the caller could change the values). Instead, copy the values into final fields.

Reply:
@electrum